gateway/api_test.go

```diff
@@ -2007,6 +2008,48 @@ func TestOAS(t *testing.T) {
 					})
 				})
 
+				// Reset
+				testUpdateAPI(t, g, &oasAPI, apiID, true)
+			})
+		})
+		t.Run("oas api/public", func(t *testing.T) {
+			apiID := oasAPIID
+			oasPubliPath := "/tyk/apis/oas/public"
+
+			t.Run("with old", func(t *testing.T) {
+
+				t.Run("get", func(t *testing.T) {
+					_, _ = g.Run(t, []test.TestCase{
```

Comment:
Fix the tests to use "ts." and if you can, move creating ts into t.Run so it's not shared. Alternatively, move all test cases into one [] value, and just do a single ts.Run over all of them. Most flakyness came from sharing ts over t.Run's.